packages/language-support/src/syntaxValidation/syntaxValidation.ts

```diff
@@ -85,25 +85,56 @@ function generateSyntaxDiagnostic(
 function detectNonDeclaredFunction(
   parsedFunction: ParsedFunction,
   functionsSchema: Record<string, Neo4jFunction>,
+  procedureSchema: Record<string, Neo4jProcedure>,
 ): SyntaxDiagnostic | undefined {
-  const lowercaseFunctionName = parsedFunction.name.toLowerCase();
+  const exists = functionExists(parsedFunction, functionsSchema);
+  if (!exists) {
+    let existsAsProcedure = false;
+    if (procedureSchema) {
+      existsAsProcedure = Boolean(procedureSchema[parsedFunction.name]);
+    }
+    if (existsAsProcedure) {
+      return generateProcedureUsedAsFunctionError(parsedFunction);
+    }
+    return generateFunctionNotFoundError(parsedFunction);
+  }
+}
+
+function functionExists(
+  functionCandidate: ParsedFunction  ,
+  functionsSchema: Record<string, Neo4jFunction>,
+): boolean {
+  if (!functionCandidate || !functionsSchema) {
+    return false;
+  }
+  const functionExistsWithExactName = Boolean(
+    functionsSchema[functionCandidate.name],
+  );
+  const lowercaseFunctionName = functionCandidate.name.toLowerCase();
   const caseInsensitiveFunctionInDatabase =
     functionsSchema[lowercaseFunctionName];
 
   // Built-in functions are case-insensitive in the database
   if (
-    caseInsensitiveFunctionInDatabase &&
-    caseInsensitiveFunctionInDatabase.isBuiltIn
+    functionExistsWithExactName ||
+    (caseInsensitiveFunctionInDatabase &&
+      caseInsensitiveFunctionInDatabase.isBuiltIn)
   ) {
-    return undefined;
+    return true;
+  } else {
+    return false;
   }
+}
 
-  const functionExistsWithExactName = Boolean(
-    functionsSchema[parsedFunction.name],
+function generateFunctionUsedAsProcedureError(
+  parsedProcedure: ParsedProcedure,
+): SyntaxDiagnostic {
+  return generateSyntaxDiagnostic(
+    parsedProcedure.rawText,
+    parsedProcedure,
+    DiagnosticSeverity.Error,
+    `Procedure ${parsedProcedure.name} is not present in the database. Did you mean to call the function ${parsedProcedure.name}? Only procedures can be called inside a CALL clause.`,
```

Comment:
It would be better if the error was actionable IMHO:

```
${parsedProcedure.name} is a function, not a procedure. Did you mean to use the function ${parsedProcedure.name} with a RETURN clause instead of a CALL clause?
```